src/compat/db_admin.c

```diff
@@ -2754,12 +2754,38 @@ db_set_system_parameters (const char *data)
       rc = sysprm_change_server_parameters (assignments);
     }
 
-  if (rc == PRM_ERR_NO_ERROR)
+  if (rc == PRM_ERR_NO_ERROR || rc == PRM_ERR_NOT_FOR_CLIENT || rc == PRM_ERR_NOT_FOR_CLIENT_NO_AUTH)
     {
       /* values were successfully set on server, set them on client too */
       sysprm_change_parameter_values (assignments, true, true);
     }
 
+  for (SYSPRM_ASSIGN_VALUE * ptr = assignments; ptr != NULL; ptr = ptr->next)
+    {
+      if (ptr->prm_id == PRM_ID_LK_TIMEOUT)
+	{
+	  SYSPRM_ASSIGN_VALUE *tmp;
+
+	  rc = sysprm_obtain_parameters ((char *) prm_get_name (PRM_ID_LK_TIMEOUT), &tmp);
+	  if (tmp->value.i >= 0)
+	    {
+	      tran_reset_wait_times (tmp->value.i * 1000);
+	    }
+	  else
+	    {
+	      tran_reset_wait_times (tmp->value.i);
+	    }
```

Comment:
1. PRM_ID_LK_TIMEOUT는 이미 msec 단위이기 때문에, `* 1000`이 필요하지 않습니다.
2. 이에 따라, `if (tmp->value.i >= 0)` 조건 검사 역시 필요하지 않습니다.
3. 추가적으로, 확인 결과 유효하지 않은 범위의 값 (ex) set system parameters 'lock_timeout=-2';) 수행 시 sysprm_validate_change_parameters() 함수에서 PRM_ERR_BAD_RANGE 에러를 발생시켜 에러 처리되고 있습니다.
```suggestion
	  rc = sysprm_obtain_parameters ((char *) prm_get_name (PRM_ID_LK_TIMEOUT), &tmp);
	 
	  tran_reset_wait_times (tmp->value.i);
```

Reply:
PRM_ID_LK_TIMEOUT 설정값에 단위를 입력하지 않으면 ms로 간주되지만 내부 함수에서 s 단위로 보정합니다. 그래서 보정된 값을 가지고 오기 위해 sysprm_obtain_parameters() 를 사용하였습니다. 얻은 온 보정값은 s(second)로 보정된 값이여서 값이 있는 경우에는 *1000 한 것입니다.
infinite 표현한 값은 -1 이고, 나머지 음수는 오류를 낼 것 같네요.

Reply:
추가로 ;set lock_timeout=default을 확인하려면 sysprm_obtain_parameters() 사용해서 변경된 값을 읽어와야 합니다.